h/static/scripts/annotator/plugin/enhancedanchoring.coffee

```diff
@@ -83,6 +89,37 @@ class Anchor
       # Kill the list if it's empty
       delete @anchoring.anchors[index] unless anchors.length
 
+  # Scroll to this anchor
+  scrollTo: ->
+    currentPage = @anchoring.document.getPageIndex()
+
+    if @startPage is @endPage and currentPage is @startPage
+      # It's all in one page. Simply scrolling
+      @highlight[@startPage].scrollTo()
+    else
+      if currentPage < @startPage
+        # We need to go forward
+        wantedPage = @startPage
+        scrollPage = wantedPage - 1
+      else if currentPage > @endPage
+        # We need to go backwards
+        wantedPage = @endPage
+        scrollPage = wantedPage + 1
+      else
+        # We have no idea where we need to go.
+        # Let's just go to the start.
+        wantedPage = @startPage
+        scrollPage = wantedPage
+
+      # Is this rendered?
+      if @anchoring.document.isPageMapped wantedPage
+        # The wanted page is already rendered, we can simply go there
+        @highlight[wantedPage].scrollTo()
+      else
+        # Not rendered yet. Go to the page, we will continue from there
+        @pendingScroll = wantedPage
```

Comment:
`pendingScroll` reads like a boolean. If this is a string that is intended to be used as a key in `@highlight` perhaps `pendingPage` would be better.